### ISSUE TITLE
Error with CUDA device IDs

### ISSUE THREAD

**Issue by wilke** [JDACS4C-IMPROVE/DRPreter#5]
Command: ```
time singularity exec --nv --bind `pwd`/tmp/:/candle_data_dir build/DRPreter.sif train.sh 2 /candle_data_dir/```

Try with 0, 1, and 2 : GPUID 0 produces no error

Output:
```
  File "/opt/conda/lib/python3.10/site-packages/torch/nn/modules/module.py", line 797, in _apply
    module._apply(fn)
  [Previous line repeated 2 more times]
  File "/opt/conda/lib/python3.10/site-packages/torch/nn/modules/module.py", line 820, in _apply
    param_applied = fn(param)
  File "/opt/conda/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1143, in convert
    return t.to(device, dtype if t.is_floating_point() or t.is_complex() else None, non_blocking)
RuntimeError: CUDA error: invalid device ordinal
CUDA kernel errors might be asynchronously reported at some other API call, so the stacktrace below might be incorrect.
For debugging consider passing CUDA_LAUNCH_BLOCKING=1.
Compile with `TORCH_USE_CUDA_DSA` to enable device-side assertions.
```

_Originally posted by @wilke in https://github.com/JDACS4C-IMPROVE/DRPreter/issues/4#issuecomment-1727867448_

**Comment by mcnaughtonadm** [comment]
Issue is with how PyTorch names available devices in code different than how CUDA_VISIBLE_DEVICES is set. Fix is is 54cdf60 on develop.

**Comment by wilke** [comment]
Status: **OK**

Building image:

```bash
time singularity build --fakeroot build/DRPreter.sif definitions/DRPreter.def
...
+ mkdir /candle_data_dir
INFO:    Adding labels
INFO:    Adding environment to container
INFO:    Creating SIF file...
INFO:    Build complete: build/DRPreter.sif
```

Testing build:

```bash
time singularity run --nv --bind `pwd`/tmp/:/candle_data_dir build/DRPreter.sif train.sh 7 /candle_data_dir --epochs 1
...
-------- DRPreter -------
sim: False
##### Seed: 42 #####
                MSE     RMSE    MAE     PCC     SCC
Train result: 1.6790000200271606        1.295799970626831       0.9745  0.8897  0.8542
Val result: 1.6943000555038452  1.3015999794006348      0.973   0.8863  0.8511
Test result: 1.740399956703186  1.319200038909912       0.9951  0.8847  0.8493

IMPROVE_RESULT val_loss:        {'val_loss': 1.6943000555038452, 'mae': 0.973, 'scc': 0.8511, 'pcc': 0.8863, 'rmse': 1.3015999794006348}
```
